src/Test/Tasty/MGolden.hs

```diff
@@ -100,8 +100,8 @@ printDetails Golden{..} expected actual = ResultDetailsPrinter print
         printDiff :: Diff.Diff Text -> IO ()
         printDiff = \case
           (Diff.Both   line _) -> printLine ' ' neutralFormat line
-          (Diff.First  line)   -> printLine '+' addFormat line
           (Diff.Second line)   -> printLine '-' removeFormat line
+          (Diff.First  line)   -> printLine '+' addFormat line
```

Comment:
@snusnu Unfortunately this approach does not work. We receive a list of `Diff` types that are of either constructor, and the order the lines are printed are not affected by the order of the pattern matches.